client/src/main/java/com/aws/greengrass/cli/adapter/impl/NucleusAdapterIpcClientImpl.java

```diff
@@ -209,6 +230,213 @@ public CreateDebugPasswordResponse createDebugPassword() {
+        PublishToTopicRequest publishToTopicRequest = new PublishToTopicRequest();
+        PublishMessage publishMessage = new PublishMessage();
+        BinaryMessage binaryMessage = new BinaryMessage();
+        binaryMessage.setMessage(message.getBytes(StandardCharsets.UTF_8));
+        publishMessage.setBinaryMessage(binaryMessage);
+        publishToTopicRequest.setPublishMessage(publishMessage);
+        publishToTopicRequest.setTopic(topicName);
+        PublishToTopicResponseHandler responseHandler = getIpcClient().publishToTopic(publishToTopicRequest, Optional.empty());
+        CompletableFuture<PublishToTopicResponse> response = responseHandler.getResponse();
+        try {
+            response.get();
+        } catch (InterruptedException interruptedException) {
+            System.out.println("publish interrupted.");
+        } catch (ExecutionException e) {
+            System.err.println("Exception occurred when publishing to topic.");
+            System.exit(1);
+        } finally {
+            close();
+        }
+    }
+
+    @Override
+    public void publishToIoTCore(String topicName, String message, String qos) {
+        QOS qos1 = QOS.get(qos);
+        PublishToIoTCoreRequest publishToIoTCoreRequest = new PublishToIoTCoreRequest();
+        publishToIoTCoreRequest.setTopicName(topicName);
+        publishToIoTCoreRequest.setPayload(message.getBytes(StandardCharsets.UTF_8));
+        publishToIoTCoreRequest.setQos(qos1);
+        PublishToIoTCoreResponseHandler responseHandler = getIpcClient().publishToIoTCore(publishToIoTCoreRequest, Optional.empty());
+        CompletableFuture<PublishToIoTCoreResponse> response = responseHandler.getResponse();
+        try {
+            response.get();
+        } catch (InterruptedException interruptedException) {
+            System.out.println("publish interrupted.");
+        } catch (ExecutionException e) {
+            System.err.println("Exception occurred when publishing to IoT Core.");
+            System.exit(1);
+        } finally {
+            close();
+        }
+    }
+
+    @Override
+    public void subscribeToTopic(String topic) {
+        try {
+            StreamResponseHandler<SubscriptionResponseMessage> streamResponseHandler =
+                    new SubscriptionResponseHandler(topic);
+            SubscribeToTopicResponseHandler responseHandler =
+                    subscribeToTopic(getIpcClient(), topic, streamResponseHandler);
+            CompletableFuture<SubscribeToTopicResponse> futureResponse =
+                    responseHandler.getResponse();
+            try {
+                futureResponse.get();
+                System.out.println("Successfully subscribed to topic: " + topic);
+            } catch (ExecutionException e) {
+                if (e.getCause() instanceof UnauthorizedError) {
+                    System.err.println("Unauthorized error while publishing to topic: " + topic);
```

Comment:
This is a subscription, but your error message says `while publishing to topic`

Reply:
Also, why do you need to handle `UnauthorizedError` just here but not for other operations?